platform/CircularBuffer.h

```diff
@@ -154,6 +154,22 @@ class CircularBuffer {
         core_util_critical_section_exit();
         return elements;
     }
+
+    /** Peek into circular buffer without popping
+     *
+     * @param data Data to be peeked from the buffer
+     * @return True if the buffer is not empty and data contains a transaction, false otherwise
+     */
+    bool peek(T& data) const {
```

Comment:
Hmm, does data need to be const if peek is a const member function?

Reply:
const member function is not allowed to modify the object elements, which is not done in case of this API. But we are passing data as return value hence it cannot be const.

Reply:
That would be an issue if it was a const reference (`T& const data`). But since it's a reference-to-const (`const T& data`), it should be fine.

Does this compile as is with a const CircularBuffer?

Reply:
Not all members of CircularBuffer are const, without this API also it won't compile as const CircularBuffer.

`(const T& data)` in case of template class says value of data variable cannot be modified in the function. We are passing the data value, hence it cannot be const. I tried compilation with const and it failed.

Reply:
Oh, that's annoying if there's a similar issue in other functions.

`const T& data` can be assigned with issue.